src/main/java/com/microsoft/sqlserver/jdbc/SQLServerParameterMetaData.java

```diff
@@ -330,7 +330,7 @@ private String escapeParse(StringTokenizer st,
         String fullName;
         nameFragment = firstToken;
         // skip spaces
-        while (" ".equals(nameFragment) && st.hasMoreTokens()) {
+        while ((0 == nameFragment.trim().length()) && st.hasMoreTokens()) {
```

Comment:
Yes, currently `firstToken` cannot be `null`, but there might be more caller methods in the future :)